src/pybind/mgr/volumes/mds-monitor.py

```diff
@@ -0,0 +1,49 @@
+from threading import Event, Thread
+import errno
+try:
+    import queue as Queue
+except ImportError:
+    import Queue
+
+class MDSMonitor(Thread):
```

Comment:
I'm not really sure we want to implement automatic MDS creation at least initially. I prefer that the orchestrator (rook) take care of bringing/up down MDSes on request from mgr, and ensures that the MDS cluster state remains as desired.

Reply:
MDS creation for volumes is tricky because you can't assign an MDS to a particular file system so "ownership" cannot be established. The volume manager might make an MDS for a file system but it could join standby-replay for another. As in Rook/Kubernetes, it is often better to define the intended state of the system (this file system should have max_mds=2 and standby-replay daemons) and then work to achieve that state.

So I would suggest either two philosophies: (a) let the orchestrator be responsible for making sure the file systems are not degraded due to lack of MDSs or (b) delegate that responsibility to the mgr by having it tell the orchestrator to spawn more MDSs when the FSs are degraded. (b) is more attractive because we only do this logic in one location.

Reply:
(b) seems simpler and more robust!  I don't like the idea of orchestrators having to understand or pay attention to fs states at all.

Reply:
The Rook orchestrator sets `"onlyManageDaemons": True,` in the CRD, which makes (b) the existing philosophy already. I'd also favor to keep it this way.